registry/types.go

```diff
@@ -28,11 +45,21 @@ type ImgData struct {
 	Tag             string `json:",omitempty"`
 }
 
-type RegistryInfo struct {
-	Version    string `json:"version"`
-	Standalone bool   `json:"standalone"`
+// PingResult contains the information returned when pinging a registry. It
+// indicates the registry's version and whether the registry claims to be a
+// standalone registry.
+type PingResult struct {
+	// Version is the registry version supplied in the
+	// X-Docker-Registry-Version header
```

Comment:
The header name may be decoupled from the version. For example, in version 2, we use `Docker-Registry-Api-Version` or something like that.